strider/server.py

```diff
@@ -465,6 +465,9 @@ async def lookup(
             # Update the results
             output_results.update(result_message.results)
 
+    for result in output_results:
+        result.combine_analyses_by_resource_id()
```

Comment:
This function is gonna get called a lot when there are lots of results. Can we reduce the costs here by short cutting if this call is even required?

Reply:
Max and I talked about this, and we couldn't think of a good way to reduce the load here since the results need to be checked for matching analyses after all of the results are collected. We have to support a use case where the analyses are kept separate, otherwise we could just do it as part of the update function in the line above.

Reply:
I guess you could check to see if there's only one analysis and save a function call sometimes, but I'd want to profile it to see if it matters.

Reply:
We could do this. I suspect it wouldn't really have much of an effect since the bottleneck is looping through all of the results, but it certainly wouldn't hurt.

Reply:
Maybe the analyses can instead be combined on addition to the same result or when results are merged?

Reply:
Just to throw some metrics in here, I've run a query with 8901 results through without any combining, with combining on all results, and then with the analysis length check, and they all finished in the same amount of time, so right now I'm not too worried about this.

Reply:
Thanks Max. I think that means we'll probably be safe merging this in for now then. We can revisit other strategies if we think it's a bottleneck in the future.